examples/AutoScrollExample.js

```diff
@@ -0,0 +1,157 @@
+/**
+ * Copyright Schrodinger, LLC
+ */
+
+"use strict";
+
+const FakeObjectDataListStore = require('./helpers/FakeObjectDataListStore');
+const { ImageCell, LinkCell } = require('./helpers/cells');
+const { Table, Column, Cell } = require('fixed-data-table-2');
+const React = require('react');
+
+class AutoScrollExample extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      dataList: new FakeObjectDataListStore(1000000),
+      scrollTop: 0,
+      scrollLeft: 0,
+      autoScrollEnabled: true,
+      horizontalScrollDelta: 0,
+      verticalScrollDelta: 0,
+    };
+
+    this.columns = [];
+    const cellRenderer = ({ columnKey, rowIndex }) =>
+      (<div className='autoScrollCell'> {rowIndex}, {columnKey} </div>);
+
+    for (let i = 0; i < 100; i++) {
+      this.columns[i] = (
+        <Column
+          key={i}
+          columnKey={i}
+          header={<div> {i} </div>}
+          cell={cellRenderer}
+          width={100}
+          allowCellsRecycling={true}
+        />
+      )
+    }
+
+    this.onVerticalScroll = this.onVerticalScroll.bind(this);
+    this.onHorizontalScroll = this.onHorizontalScroll.bind(this);
+    this.toggleAutoScroll = this.toggleAutoScroll.bind(this);
+    this.setHorizontalScrollDelta = this.setHorizontalScrollDelta.bind(this);
+    this.setVerticalScrollDelta = this.setVerticalScrollDelta.bind(this);
+  }
+
+  componentDidMount() {
+    setInterval(() => {
+      if (!this.state.autoScrollEnabled) {
+        return;
+      }
+      this.setState((prevState) => ({
+        scrollTop: prevState.scrollTop + (prevState.verticalScrollDelta || 0),
+        scrollLeft: prevState.scrollLeft + (prevState.horizontalScrollDelta || 0),
+      }));
+    }, 16);
+  }
+
+  render() {
+    return (
+      <div className='autoScrollContainer'>
+        {this.renderControls()}
+        {this.renderTable()}
+      </div>
+    );
+  }
+
+  renderControls() {
+    return (
+      <div className='autoScrollControls'>
+        <label>
+          Auto Scroll Enabled
+          <input type='checkbox' checked={this.state.autoScrollEnabled} onChange={this.toggleAutoScroll} />
+        </label>
+        <label>
+          Horizontal Scroll Delta
+          <input type='number' value={this.state.horizontalScrollDelta} onChange={this.setHorizontalScrollDelta} />
+        </label>
+        <label>
+          Vertical Scroll Delta
+          <input type='number' value={this.state.verticalScrollDelta} onChange={this.setVerticalScrollDelta} />
+        </label>
+      </div>
+    )
+  }
+
+  renderTable() {
+    var { dataList, scrollLeft, scrollTop } = this.state;
+    return (
+      <Table
+        rowHeight={50}
+        headerHeight={50}
+        rowsCount={dataList.getSize()}
+        width={1000}
+        height={500}
+        scrollLeft={scrollLeft}
+        scrollTop={scrollTop}
+        onVerticalScroll={this.onVerticalScroll}
```

Comment:
these are needed since the user can also scroll the table directly outside of our "autoscroll" logic